test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1244,3 +1246,23 @@
             (is (= change-perm (get-file-perms tmp-file)))
             (fs/delete tmp-file)
             (recur (nthnext perms 2))))))))
+
+(deftest create-csr-attrs-exts-test
+  (testing "when parsing a csr_attributes file"
+    (testing "and the file exists"
+      (testing "and it has non-whitelisted OIDs"
+        (let [extensions (create-csr-attrs-exts (csr-attributes-file "csr_attributes_with_auth.yaml"))
+              expected (concat attribute-file-extensions
+                               [{:oid "1.3.6.1.4.1.34380.1.3.1" ;; :pp_authorization
+                                 :critical false
+                                 :value "true"}
+                                {:oid "1.3.6.1.4.1.34380.1.3.13" ;; :pp_auth_role
+                                 :critical false
+                                 :value "com"}])]
+          (is (= (set extensions) (set expected)))))
+      (testing "and it has whitelisted OIDs"
+        (let [extensions (create-csr-attrs-exts (csr-attributes-file "csr_attributes.yaml"))]
+          (is (= (set extensions) (set attribute-file-extensions))))))
```

Comment:
seems like the key point of this test is that the new OIDs can be used by shortname (instead of OID) in csr_attributes - would be good to capture that in a `testing` string or something like that.